src/Magnum/Shaders/FlatGL.h

```diff
@@ -602,6 +603,16 @@ template<UnsignedInt dimensions> class MAGNUM_SHADERS_EXPORT FlatGL: public GL::
          */
         explicit FlatGL(NoCreateT) noexcept: GL::AbstractShaderProgram{NoCreate} {}
 
+        class CompileState;
+
+        explicit FlatGL(CompileState&& cs);
+
+        static CompileState compile(Flags flags
+        #ifndef MAGNUM_TARGET_GLES2
+        , UnsignedInt materialCount, UnsignedInt drawCount
+        #endif
+        );
```

Comment:
I think a convenience 

```cpp
        #ifndef MAGNUM_TARGET_GLES2
        static CompileState compile(Flags flags) {
            return compile(flags, 1, 1);
        }
        #endif
```

for non-UBO use cases could make sense as well, to match the similar convenience constructor.